core/metrics-core-service/src/test/java/org/hawkular/metrics/core/util/GCGraceSecondsManagerITest.java

```diff
@@ -0,0 +1,150 @@
+import org.hawkular.metrics.schema.SchemaService;
+import org.hawkular.metrics.sysconfig.ConfigurationService;
+import org.hawkular.rx.cassandra.driver.RxSession;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ResultSet;
+
+import rx.Subscription;
+import rx.subjects.PublishSubject;
+
+/**
+ * @author jsanda
+ */
+public class GCGraceSecondsManagerITest extends BaseITest {
+
+    private static class TestGCGraceSecondsManager extends GCGraceSecondsManager {
+
+        private Integer clusterSize;
+
+        private PublishSubject<Throwable> updatesFinished;
+
+        public TestGCGraceSecondsManager(RxSession session, String keyspace,
+                ConfigurationService configurationService) {
+            super(session, keyspace, configurationService);
+            updatesFinished = PublishSubject.create();
+            setUpdatesFinishedSubject(updatesFinished);
+        }
+
+        @Override
+        protected int getClusterSize() {
+            if (clusterSize == null) {
+                return super.getClusterSize();
+            }
+            return clusterSize;
+        }
+
+        public void setClusterSize(int clusterSize) {
+            this.clusterSize = clusterSize;
+        }
+
+        @Override
+        public void maybeUpdateGCGraceSeconds() {
+            Subscription subscription = null;
+            try {
+                CountDownLatch latch = new CountDownLatch(1);
+                AtomicReference<Throwable> exceptionRef = new AtomicReference<>();
+                subscription = updatesFinished.subscribe(t -> {
+                    exceptionRef.set(t);
+                    latch.countDown();
+                });
+                super.maybeUpdateGCGraceSeconds();
+                latch.await(10, TimeUnit.SECONDS);
+                if (exceptionRef.get() != null) {
+                    fail("Updates failed", exceptionRef.get());
+                }
+            } catch (InterruptedException e) {
+                fail("Timed out waiting for updates to complete", e);
+            } finally {
+                if (subscription != null) {
+                    subscription.unsubscribe();
+                }
+            }
+        }
+    }
+
+    private ConfigurationService configurationService;
+
+    private String keyspace = "gc_grace_test";
+
+    private TestGCGraceSecondsManager manager;
+
+    @BeforeClass
+    public void initClass() {
+        String nodeAddresses = System.getProperty("nodes", "127.0.0.1");
+        Cluster cluster = new Cluster.Builder()
+                .addContactPoints(nodeAddresses.split(","))
+                .build();
+        configurationService = new ConfigurationService();
+        configurationService.init(rxSession);
+
+        SchemaService schemaService = new SchemaService();
+        schemaService.run(session, keyspace, Boolean.valueOf(System.getProperty("resetdb", "true")));
```

Comment:
Is it a good idea to reset the schema midway through this test set? Any possible side effects?